cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,84 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.visit('/');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('', () => {
+  it('should register a new student', () => {
+    cy.findByPlaceholder('First Name')
+      .type(user.firstName);
+
+    cy.findByPlaceholder('Last Name')
+      .type(user.lastName);
+
+    cy.get('#userEmail').type(user.email);
+
+    cy.contains('.custom-control-label', user.gender)
+      .click();
+
+    cy.findByPlaceholder('Mobile Number')
+      .type(user.mobileNumber);
+
+    cy.get('#dateOfBirthInput')
+      .type('{selectAll}15 Sep 2005{enter}');
+
+    cy.get('.subjects-auto-complete__value-container')
+      .type('m{enter}' + 'e{enter}' + 'p{enter}');
+
+    cy.contains('.custom-control-label', user.hobby)
+      .click();
+
+    cy.findByPlaceholder('Current Address')
+      .type(user.address);
+
+    cy.get('#state')
+      .type('{downArrow}{enter}');
+
+    cy.get('#city')
+      .type('{downArrow}{enter}');
+
+    cy.get('#submit')
+      .click();
+
+    cy.contains('.modal-content', 'Thanks for submitting the form');
+
+    cy.contains('tr', 'Student Name')
+      .should('contain', user.firstName);
+
+    cy.contains('tr', 'Student Name')
+      .should('contain', user.lastName);
+
+    cy.contains('tr', 'Student Email')
+      .should('contain', user.email);
+
+    cy.contains('tr', 'Gender')
+      .should('contain', user.gender);
+
+    cy.contains('tr', 'Mobile')
+      .should('contain', user.mobileNumber);
+
+    cy.contains('tr', 'Date of Birth')
+      .should('contain', '15 September,2005');
```

Comment:
The date of birth asserted in the modal is hard-coded. It should be dynamically obtained from the user object.